src/ai/backend/storage/kmanila/__init__.py

```diff
@@ -0,0 +1,272 @@
+
+    async def put_user_volume_id(self, quota_scope_id: QuotaScopeID, data: QuotaVolumeMap) -> None:
+        await self.etcd.put_prefix(f"{QUOTA_VOLUME_ID_MAP_KEY}/{quota_scope_id.scope_id}", data)  # type: ignore[arg-type]
+
+    async def get_volume_id(
+        self,
+        session: aiohttp.ClientSession,
+        quota_scope_id: QuotaScopeID,
+        auth_info: dict[str, Any],
+        *,
+        do_hard_check: bool = True,
+    ) -> str | None:
+        if (vol_id := await self.get_user_volume_id(quota_scope_id)) is not None:
+            if not do_hard_check:
+                return vol_id
+        project_id = auth_info["project_id"]
+        async with session.get(f"/d3/adm/manila/v2/{project_id}/shares/{vol_id}") as resp:
+            if resp.status == 200:
+                volume_info = (await resp.json())["share"]
+                return volume_info["id"]
+            elif resp.status // 100 == 5:
+                raise ExternalError("Cannot get data from API server")
+        return None
+
+    async def _create_volume(
+        self,
+        session: aiohttp.ClientSession,
+        quota_scope_id: QuotaScopeID,
+        options: QuotaConfig | None,
+        auth_info: dict[str, Any],
+        *,
+        name: str,
+    ) -> str:
+        if (volume_id := await self.get_volume_id(session, quota_scope_id, auth_info)) is not None:
+            return volume_id
+        project_id = auth_info["project_id"]
+        request_body = {
+            "share": {
+                "share_proto": "nfs",
+                "share_network_id": self.default_netword_id,
+                "name": name,
+                "is_public": False,
+                "size": options.limit_bytes if options is not None else DEFAULT_VOLUME_SIZE,
+                "availability_zone": "DX-DCN-CJ",
+                "share_type": "SSD",
+            }
+        }
+        async with session.post(
+            url=f"/d3/adm/manila/v2/{project_id}/shares", json=request_body
+        ) as resp:
+            if resp.status not in (200, 201, 204):
+                raise ExternalError(
+                    f"Got invalid status code when post data to API server. {resp.status = }"
+                )
+            await asyncio.sleep(0)
+
+        # Poll creation complete
+        trial = 1
+        while True:
+            log.debug(f"Poll if the volume has been created, {trial = }")
+            if (
+                volume_id := await self.get_volume_id(session, quota_scope_id, project_id)
+            ) is not None:
+                await self.put_user_volume_id(
+                    quota_scope_id, QuotaVolumeMap(volume_id=volume_id, volume_name=name)
+                )
+                return volume_id
+            await asyncio.sleep(1)
+            trial += 1
+            if trial > self.max_poll_count:
+                raise ExternalError(f"Poll trial exceeds the maximum trial count, {trial = }")
+
+    async def _create_access_control(
```

Comment:
By only reading the method name I can't get what this function actually does. Please update the name to be more self-explanatory, perhaps like `_allow_access_to_volume()`?

Reply:
Since the API spec represent it as "Creating access control", I leave a comment on this function rather than changing the name.